### ISSUE TITLE
Accessibility: Should opening an edit form for a file/folder move focus into the form?

### ISSUE THREAD

**Issue by ChloeSartorelli** [silverstripe/silverstripe-asset-admin#1619]
In `/admin/assets` in both gallery and table view, and inside the upload field "choose existing" modal, you can open a form panel by clicking on a (or pressing the enter key on a focused) file.

Doing so leaves your focus where it was. If you are navigating using keyboard, you then have to press tab a bunch of times to get to the form. If you're using a screen reader, you may not even be aware that a form is now available to you.

- Should activating the button that displays the form also move focus into the form?
- If not, is there another technique we should be using to alert screen reader users that there is now a form for them - and how to get there?

## Notes

- When opening a modal, we move focus into the modal (often into the first interactive element in the modal) - though in this case the original button can't be accessed anyway until the modal is closed.
- When popping out the sudo mode password field, we move focus into the input field - though in this case the original button is no longer present.
- W3C and other guidance doesn't seem clear on this from what I can tell - it mostly deals with navigating to other pages or changing the content on the page entirely, not adding new content to the page.
- It _seems like_ so long as the user is activating the button (i.e. on input, not on focus) and the screen reader announcement makes it clear context will/may change (the usual example is a "submit" button, but having a button say "activate edit form" or similar would do) this is sufficient.

**Comment by emteknetnz** [comment]
This was implemented in https://github.com/silverstripe/silverstripe-asset-admin/pull/1630
